File: src/lib/tauri/scripts.ts
import { invoke } from "@tauri-apps/api/core";
import type {
  BuildRemoteScriptCommandInput,
  RunLocalScriptInput,
  ScriptEntryDetail,
  ScriptEntrySummary,
  ScriptExecutionResult
} from "../../types/scripts";

export async function listScriptEntries(): Promise<ScriptEntrySummary[]> {
  return invoke<ScriptEntrySummary[]>("list_script_entries");
}

export async function getScriptEntryDetail(path: string): Promise<ScriptEntryDetail> {
  return invoke<ScriptEntryDetail>("get_script_entry_detail", { path });
}

export async function runLocalScript(
  input: RunLocalScriptInput
): Promise<ScriptExecutionResult> {
  return invoke<ScriptExecutionResult>("run_local_script", { input });
}

export async function buildRemoteScriptCommand(
  input: BuildRemoteScriptCommandInput
): Promise<string> {
  return invoke<string>("build_remote_script_command", { input });
}

export async function getScriptWorkspaceRoot(): Promise<string> {
  return invoke<string>("get_script_workspace_root");
}

File: src-tauri/src/commands/scripts.rs
use crate::{
    app_state::AppState,
    modules::models::{
        RunLocalScriptInput, ScriptEntryDetail, ScriptEntrySummary, ScriptExecutionResult,
    },
};
use tauri::State;

#[tauri::command]
pub async fn list_script_entries(
    state: State<'_, AppState>,
) -> Result<Vec<ScriptEntrySummary>, String> {
    state
        .scripts
        .list_scripts()
        .map_err(|error| error.to_string())
}

#[tauri::command]
pub async fn get_script_entry_detail(
    path: String,
    state: State<'_, AppState>,
) -> Result<ScriptEntryDetail, String> {
    state
        .scripts
        .read_script(&path)
        .map_err(|error| error.to_string())
}

#[tauri::command]
pub async fn run_local_script(
    input: RunLocalScriptInput,
    state: State<'_, AppState>,
) -> Result<ScriptExecutionResult, String> {
    state
        .scripts
        .run_local_script(input)
        .await
        .map_err(|error| error.to_string())
}

#[tauri::command]
pub async fn get_script_workspace_root(
    state: State<'_, AppState>,
) -> Result<String, String> {
    Ok(state.scripts.root_dir().to_string_lossy().into_owned())
}
